modules/discovery-zen/build.gradle

```diff
@@ -17,19 +17,17 @@
  * under the License.
  */
 
-package org.elasticsearch.discovery.zen;
-
-import org.elasticsearch.cluster.node.DiscoveryNode;
+esplugin {
+    description 'Zen discovery is the built in discovery module for Elasticsearch'
+    classname 'org.elasticsearch.discovery.zen.ZenDiscoveryPlugin'
+}
 
-import java.util.List;
+compileJava.options.compilerArgs << "-Xlint:-cast,-deprecation,-rawtypes,-try,-unchecked"
+compileTestJava.options.compilerArgs << "-Xlint:-cast,-deprecation,-rawtypes,-try,-unchecked"
 
-/**
- * A pluggable provider of the list of unicast hosts to use for unicast discovery.
- */
-public interface UnicastHostsProvider {
+// somehow both packaged version and main source set end up on classpath, tripping jarhell check
```

Comment:
Jarhell occurs because this module was added directly as a dependency for test framework. To be truly independent, we need the test framework to not depend on this.